src/firefly/js/api/ApiHighlevelBuild.js

```diff
@@ -37,20 +37,130 @@ function build(llApi) {
+        }
+    });
+
+    dispatchTableSearch(request, options);
+    renderDOM(targetDiv, TablesContainer, contProps);
+}
 
 function buildTablePart(llApi) {
-    //todo
-    return {};
+    const {dispatchTableFetch}= llApi.action;
+
+    /**
+     * @typedef {object} TblOptions    table options
+     * @prop {string}  tbl_group    the group this table belongs to.  Defaults to 'main'.
+     * @prop {boolean} removable    true if this table can be removed from view.  Defaults to true.
+     * @prop {boolean} showUnits    defaults to false
+     * @prop {boolean} showFilters  defaults to false
+     * @prop {boolean} selectable   defaults to true
+     * @prop {boolean} expandable   defaults to true
+     * @prop {boolean} showToolbar  defaults to true
+     * @prop {boolean} border       defaults to true
+     */
+
+    /**
+     * The general plotting function to plot a table.
+     * @param {string|HTMLDivElement} targetDiv to put the table in.
+     * @param {Object} request         request object created from
+     * @param {TblOptions} options     table options.
+     */
+    const showTable= (targetDiv, request, options)  => doShowTable(llApi, targetDiv, request, options);
+
+    return {showTable};
 }
 
-function buildXYandHistPart(llApi) {
-    //todo
-    return {};
+/*---------------------------- TABLE PART >----------------------------*/
+
+
+function buildChartPart(llApi) {
+
+    /**
+     * @typedef {object} XYPlotOptions  xy plot options
+     * @prop {string}  source       location of the ipac table, url or file path; ignored when XY plot view is added to table
+     * @prop {string}  QUERY_ID     required when XY plot view is added to the table. It connects this XY Plot to the table and should be the same string that you specified as the div parameter when you created the table.
+     * @prop {string}  chartTitle   title of the chart
+     * @prop {string}  xCol         column or expression to use for x values, can contain multiple column names ex. log(col) or (col1-col2)/col3
+     * @prop {string}  yCol         column or expression to use for y values, can contain multiple column names ex. sin(col) or (col1-col2)/col3
+     * @prop {number}  xyRatio      aspect ratio (must be between 1 and 10)
+     * @prop {string}  stretch      fit or fill
+     * @prop {string}  xLabel       label to use with x axis
+     * @prop {string}  yLabel       label to use with y axis
+     * @prop {string}  xUnit        unit for x axis
+     * @prop {string}  yUnit        unit for y axis
+     * @prop {string}  xOptions     comma separated list of x axis options: grid,flip,log
+     * @prop {string}  yOptions     comma separated list of y axis options: grid,flip,log
+     */
+
+    /**
+     * The general plotting function to plot an XY Plot.
+     * @param {string|HTMLDivElement} targetDiv to put the chart in.
+     * @param {XYPlotOptions} parameters the request object literal with the chart parameters
```

Comment:
thanks for using @typedef in consistent with table api.  if this does not work out, at least we can address them in the same manner.